### PR TITLE
Date and Number allowed; Semicolon after background,

### PR DESCRIPTION
**Pull request by MatthiasSaihttam**
I was just fixing some miscellaneous bugs I found, namely #450, #520, and an unreported issue where background doesn't autofill a semicolon after it. While this builds and runs fine, I have two questions before you merge this.
First, I [added an entry](https://github.com/Khan/live-editor/commit/79e36bb6666c7f7d9a45cc9a6d164fb38867d3a3) to `functionParamCount` for `Date`, in order to make BabyHint work with `new Date()`. I believe this same change should be applied to the [BabyHint repository itself](https://github.com/khan/babyhint), and would be willing to make a pull request there with the change. Second, I haven't updated pre-built version of the code with the use of `Date` or `Number`, and don't know how to do that (or if it needs to be done). I'm new at GitHub, so thanks for any help.

### CODE REVIEW

**Comment by SharkFinPro** [comment]
I hope they push this to MASTER

File: js/output/pjs/babyhint.js [hunk 1]
```diff
@@ -112,7 +112,8 @@ var BabyHint = {
         "textAlign": [1, 2],
         "textFont": [1, 2],
         "translate": [2, 3],
-        "vertex": [2, 4]
+        "vertex": [2, 4],
+        "Date": [0, 1, 2, 3, 4, 5, 6, 7],
```
Comment (kevinbarabash) [inline js/output/pjs/babyhint.js]: I didn't realize you could pass this many args to `Date`.  TIL.

**Comment by kevinbarabash** [comment]
Changes look good.  I'm a little confused as to why there are only build changes to `live-editor.tooltips.js`.  @MatthiasSaihttam could you run `gulp build` and include the build files in your pull request?

**Comment by MatthiasSaihttam** [comment]
Ok, I've updated the build files with my changes. You might want to make sure everything works locally before including it in master. Also, did you see my comment about updating the BabyHint repository?

**Comment by kevinbarabash** [comment]
@MatthiasSaihttam I did see your comments about the BabyHint repo.  I set up the BabyHint repo with the hope that someday we'd be able to replace the copy of BabyHint that's in the live-editor with the repo version.  Unfortunately, I never got around to finishing that project.

Thanks for including the build files.  Hopefully I'll have time later this week to check out the changes.  Something looks horribly wrong with the build.  At first glance it looks like some of the dependencies aren't installing correctly... not sure what's causing that to happen.

**Comment by MatthiasSaihttam** [comment]
Yes, I tried to include another commit late last night to fix #588, but it apparently messed something up. I'll work on undoing that change now.

**Comment by kevinbarabash** [comment]
@MatthiasSaihttam thanks for fixing the build.  Could you submit a separate pull request with your fix for https://github.com/Khan/live-editor/issues/588?

**Comment by MatthiasSaihttam** [comment]
I finally got the code updated to where I would like it, and now the tests are failing at Travis CI again. However I can build and the run the code locally without any problem, and I get success from all the tests when I run them on my computer. And ideas on what's going on?

**Comment by kevinbarabash** [comment]
I re-ran the test and it past.  Must have been a flakey test.

**Comment by MatthiasSaihttam** [comment]
Awesome, does that mean you can merge it?

**Comment by kevinbarabash** [comment]
@MatthiasSaihttam I jumped the gun a little bit with merging.  Could you sign our CLA?   http://www.khanacademy.org/r/cla

**Comment by MatthiasSaihttam** [comment]
Done. Now you can come to my house and watch me...

**Comment by kevinbarabash** [comment]
I'll try to do a deploy tomorrow

**Comment by MatthiasSaihttam** [comment]
I was just looking around and noticed that #450 and #520 were still open, probably because I didn't word my commit message in a way that let Github know it would fix them. If you get a chance, could you close them?

P.S. How about that deploy, eh?

**Pull request merged**